sbt-scalajs-bundler/src/main/scala/scalajsbundler/BundlingMode.scala

```diff
@@ -0,0 +1,27 @@
+package scalajsbundler
+
+/**
+  * The BundlingMode choice dictates how webpack task is set up.
+  */
+sealed trait BundlingMode
+
+object BundlingMode {
+
+  /** Bundle the whole application */
+  case object Application extends BundlingMode
+
+  /** Bundle the dependencies of the application */
+  sealed trait Library extends BundlingMode {
+    /** Name of the global variable containing the dependencies */
+    def exportedName: String
+  }
+
+  /** Bundle only the libraries used by the application. */
+
+  case class LibraryOnly(exportedName: String = defaultLibraryExportedName) extends Library
+  /** Bundle the libraries used by the application and concatenates the output of Scala.js to that bundle */
+
+  case class LibraryAndApplication(exportedName: String = defaultLibraryExportedName) extends Library
+
+  val defaultLibraryExportedName = "ScalaJSBundlerLibrary"
+}
```

Comment:
That would be awesome if you could expand a little bit the scaladoc to explain the impact of each bundling mode on the generated Webpack config file (e.g. which properties are set in which mode?)

Reply:
Your changes are great! I really love all the cross links you added :)